packages/twenty-server/src/engine/core-modules/file/services/file-metadata.service.ts

```diff
@@ -0,0 +1,60 @@
+import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+
+import { Repository } from 'typeorm';
+
+import { CreateFileDTO } from 'src/engine/core-modules/file/dtos/create-file.dto';
+import { FileDTO } from 'src/engine/core-modules/file/dtos/file.dto';
+import { FileEntity } from 'src/engine/core-modules/file/entities/file.entity';
+import { extractRelativePath } from 'src/engine/core-modules/file/utils/extract-relative-path.utils';
+
+import { FileService } from './file.service';
+
+@Injectable()
+export class FileMetadataService {
+  constructor(
+    @InjectRepository(FileEntity, 'core')
+    private readonly fileRepository: Repository<FileEntity>,
+    private readonly fileService: FileService,
+  ) {}
+
+  async createFile(fileData: CreateFileDTO): Promise<FileDTO> {
+    const file = this.fileRepository.create(fileData);
+    const savedFile = await this.fileRepository.save(file);
+
+    return savedFile;
+  }
+
+  async deleteFileById(
+    id: string,
+    workspaceId: string,
+  ): Promise<FileDTO | null> {
+    const file = await this.fileRepository.findOne({
+      where: { id, workspaceId },
+    });
+
+    if (!file) {
+      return null;
+    }
+
+    try {
+      const relativePath = extractRelativePath(file.fullPath);
```

Comment:
It feels like we shouldn't store the http://xxx url in the database, this is something that's going to be invariant (repeated 1M+ times if we have 1M+ rows) and it's better to build it dynamically when you need it. Maybe then you won't need this extractRelativePath